### DISCUSSION TITLE
Welcome to Fooocus Discussions!

### DISCUSSION THREAD

**Discussion by mashb1t** [mashb1t/Fooocus#12] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by uiyo** [comment]
Hi there, 😀
I noticed that Fooocus is currently using Gradio 3.41.2, while the latest version is 4.x.x. I've been having some network connectivity issues with Fooocus, and I've found that similar tools using the latest Gradio don't have these problems.

I've tried upgading Gradio or modifying the Gradio's connection related files, but that requires modifying the Fooocus source code, which I'd rather not do.

I really like Fooocus and think it's a great tool, so I was wondering if you'd consider using latest 4.x.x version Gradio in Fooocus.

**Comment by mashb1t** [comment]
https://github.com/lllyasviel/Fooocus/pull/961
https://github.com/lllyasviel/Fooocus/discussions/1534

There were already attempts but they've never been completed. Frankly speaking it's not the highest priority, but in general packages have to be updated sooner or later. Fooocus is also using Pillow 9.2.0, where v10 has been out for a while now etc.
Feel free to support in the upgrade, i'll update the PR https://github.com/lllyasviel/Fooocus/pull/961 if possible soon.

**Reply by uiyo** [reply]
great🎉, thank you!👍

**Comment by blckbx** [comment]
Thank you for picking up this great piece of software and advancing its abilities!

**Comment by uiyo** [comment]
Hi, me again :)
Just wanted to suggest checking out [Align Your Steps (nvidia.com)](https://research.nvidia.com/labs/toronto-ai/AlignYourSteps/) for sampling optimization. It's a pretty cool algorithm that could help improve image quality and efficiency, maybe better than FreeU.

**Comment by DIGITAL-SURFER** [comment]
Hi. I am an UK based illustrator and designer, who unlike others actually embraces AI and sees the benefits of working in a different way,  Started off with Easy Diffusion and A1111, recently upgraded my setup to use SDXL, only to find A1111 'spikes' and crashes with SDXL models.  From there I found the Forge variant developed by the same guy that developed Fooocus, so eventually found my way here. 

The one thing I really love about Fooocus, is it has the only inpainting that works well.  I came across this particular fork, which is apart from the fact it still being developed, also introduced masking.  Only issue I have with this option sometimes, is finding an appropriate trigger word to mask what I need. I have found using the mask function in 'Inpaint Anything' with its segment display is far more usable, but am guessing the structure means it would be difficult to implement in Fooocus?

Thanks for your hard work in keeping Fooocus alive and still out there. None of us want to be tied to corporate online solutions like Midjourney etc.

**Reply by mashb1t** [reply]
@DIGITAL-SURFER inpaint anything is based on StableDiffusionInpaintPipeline (transformers) and as Fooocus still uses Gradio 3.41.2 we can't use segment map layers as they afaik are only supported from 4.x.
The mask generation functionality will be included in a future version of Fooocus (core), milestone package updates, but i just need a bit more time for this.

**Reply by DIGITAL-SURFER** [reply]
@mashb1t  Thanks for the clarification on that. 

Incidentally, the recent hyper-sdxl preset brings in images in half the time that the Forge variant of A1111 does, so that has been a really worthwhile addition to Fooocus.   Everything is going in a good direction.
